controllers/dspipeline_params.go

```diff
@@ -383,11 +390,11 @@ func (p *DSPAParams) SetupObjectParams(ctx context.Context, dsp *dspa.DataScienc
 
 }
 
-func (p *DSPAParams) SetupMLMD(ctx context.Context, dsp *dspa.DataSciencePipelinesApplication, client client.Client, log logr.Logger) error {
+func (p *DSPAParams) SetupMlmdV1() error {
```

Comment:
I don't think we should split this out as it would lead to messy structure with 90% duplicated code.  Instead, we should handle deploying MLMDWriter only if using v1 pipelines (there is a helper function already in here that will help determine this condition)

Reply:
Done.